lib/ansible/galaxy/collection.py

```diff
@@ -461,6 +522,60 @@ def validate_collection_name(name):
                        "characters from [a-zA-Z0-9_] only." % name)
 
 
+def verify_collections(collections, search_path, apis, validate_certs, ignore_errors):
+    existing_collections = _find_existing_collections(search_path)
+
+    current_collections = dict(('%s.%s' % (collection.namespace, collection.name), collection) for collection in existing_collections)
+
+    with _display_progress():
+        with _tempdir() as b_temp_path:
+            for collection in collections:
+                try:
+
+                    b_temp_tar_path = None
+                    b_collection = to_bytes(collection[0], errors='surrogate_or_strict')
+                    located_by_name = False
+
+                    if os.path.isfile(b_collection):
+                        remote_collection = CollectionRequirement.from_tar(to_bytes(collection[0]), False, parent=None)
+                    elif os.path.isdir(b_collection):
+                        remote_collection = CollectionRequirement.from_path(b_collection, False, parent=None)
+                    elif urlparse(collection[0]).scheme.lower() in ['http', 'https']:
+                        b_temp_tar_path = _download_file(collection[0], b_temp_path, None, validate_certs)
+                        remote_collection = CollectionRequirement.from_tar(b_temp_tar_path, False, parent=None)
+                    else:
+                        located_by_name = True
+                        remote_collection = CollectionRequirement.from_name(collection[0], apis, collection[1], False, parent=None)
+
+                    collection_name = '%s.%s' % (remote_collection.namespace, remote_collection.name)
+                    collection_version = remote_collection.latest_version
+
+                    if b_temp_tar_path is None:
+                        if not located_by_name:
+                            remote_collection = CollectionRequirement.from_name(collection_name, apis, collection_version, False, parent=None)
+                        remote_collection._get_metadata()
```

Comment:
This may impact @samdoran as well, but if this is useful outside of the `CollectionRequirement` or not called automatically, then we might want to consider removing the leading `_`